### PR TITLE
app-editors/pyvim version bump and improvements

### PR DESCRIPTION
**Pull request by CyberTailor**

### CODE REVIEW

**Review by MatthewGentoo — COMMENTED** [review]
Is it worth keeping the stable version 3.0.2 around until 3.0.3 can be marked stable? :)

**Comment by CyberTailor** [comment]
OK, let's leave it. Updated the PR.

File: app-editors/pyvim/pyvim-3.0.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+PYTHON_COMPAT=( python3_{8..11} )
+DISTUTILS_USE_PEP517=setuptools
+inherit distutils-r1 edo
+
+COMMIT="14118ad2e4d0da2e955fd9069b8772408307618b"
+DESCRIPTION="An implementation of Vim in Python"
+HOMEPAGE="https://pypi.org/project/pyvim/ https://github.com/prompt-toolkit/pyvim"
+SRC_URI="
+	!test? ( mirror://pypi/${PN::1}/${PN}/${P}.tar.gz )
+	test? ( https://github.com/prompt-toolkit/${PN}/archive/${COMMIT}.tar.gz -> ${P}.gh.tar.gz )
```
Comment (thesamesam) [inline app-editors/pyvim/pyvim-3.0.3.ebuild:15]: Can't you unconditionally use the GH tarball...?

**Comment by CyberTailor** [comment]
But it's not versioned.

**Comment by thesamesam** [comment]
> But it's not versioned.

So? The thing you're testing isn't the thong users get.

File: app-editors/pyvim/pyvim-3.0.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..11} )
+DISTUTILS_USE_PEP517=setuptools
+inherit distutils-r1 edo
+
+COMMIT="14118ad2e4d0da2e955fd9069b8772408307618b"
+DESCRIPTION="An implementation of Vim in Python"
+HOMEPAGE="https://pypi.org/project/pyvim/ https://github.com/prompt-toolkit/pyvim"
+SRC_URI="https://github.com/prompt-toolkit/${PN}/archive/${COMMIT}.tar.gz -> ${P}.gh.tar.gz"
```
Comment (laumann) [inline app-editors/pyvim/pyvim-3.0.3.ebuild:13]: I think you could drop the `.gh` part of the file extension.

Comment (thesamesam) [reply app-editors/pyvim/pyvim-3.0.3.ebuild:13]: Doesn't pkgcheck require this now?

Comment (laumann) [reply app-editors/pyvim/pyvim-3.0.3.ebuild:13]: You are right @thesamesam (https://github.com/pkgcore/pkgcheck/pull/383) sorry for the noise @CyberTailor! TIL.

**Comment by CyberTailor** [comment]
OK, I'm convinced. Updated the PR.

**Pull request closed without merging**